Preview docs for CloudPrem (#28464)
DataDog/documentation: content/en/cloudprem/processing.md

You configure log processing pipelines using a JSON file that adheres to the same format as Datadog pipeline configurations.

## Setting up processing

1. (Optional) If you have existing pipelines in Datadog, you can fetch the configuration using the [Logs Pipelines API][2]:

   ```bash
   curl -X GET "https://api.datadoghq.com/api/v1/logs/config/pipelines" \
    -H "Accept: application/json" \
    -H "DD-API-KEY: ${DD_API_KEY}" \
    -H "DD-APPLICATION-KEY: ${DD_APP_KEY}" > my-pipelines_config.json
   ```

   This JSON file can be used directly with CloudPrem.

2. To set the configuration in the Helm Chart, provide the path to your JSON configuration file using the `pipelinesConfig` parameter in the CloudPrem Helm chart:

   ```bash
   helm repo update

>    helm upgrade <release-name> -n <namespace> --set-file pipelinesConfig=./pipelines_config.json -f <custom_values_file.yaml>

Reviewer: ```suggestion
   helm upgrade <RELEASE_NAME> -n <NAMESPACE> --set-file pipelinesConfig=./pipelines_config.json -f your_custom_values_file.yaml
```

   ```

   CloudPrem logs an informational message (`Successfully read pipeline config file`) when it successfully reads the configuration file. Any processors defined in the file that are not supported by CloudPrem are ignored during startup.
   **Note**: Helm imposes a 1 MB size limit on the configuration file due to its underlying etcd storage.

## Configuration file format

The configuration is a JSON array, where each element represents a processor or a nested pipeline.  

The order of elements in the array defines the sequential execution order of the processors.  The structure mirrors the output of the Datadog API endpoint `api/v1/logs/config/pipelines`.

```json
[
 {...processor1},
 {...processor2}
]
```

> ```json

Reviewer: For these json codeblocks, typically each indentation level would be 2 spaces (or 4, but usually 2). Not sure how much it matters to you, it just looks a bit odd if you're used to working with JSON. You can probably just run it through a JSON formatter if you want to fix it quickly.

                        "supportRules": "",
                        "matchRules": "agent_rule %{date(\"yyyy-MM-dd HH:mm:ss z\"):timestamp} \\| %{notSpace:agent} \\| %{word:level} \\| \\(%{notSpace:filename}:%{number:lineno} in %{word:process}\\) \\|( %{data::keyvalue(\":\")} \\|)?( - \\|)?( \\(%{notSpace:pyFilename}:%{number:pyLineno}\\) \\|)?%{data}\nagent_rule_pre_611 %{date(\"yyyy-MM-dd HH:mm:ss z\"):timestamp} \\| %{word:level} \\| \\(%{notSpace:filename}:%{number:lineno} in %{word:process}\\)%{data}\njmxfetch_rule     %{date(\"yyyy-MM-dd HH:mm:ss z\"):timestamp} \\| %{notSpace:agent} \\| %{word:level}\\s+\\| %{word:class} \\| %{data}"
                    }
                },
                {
                    "id": "xnsd5oanXq2893utcsQ",
                    "type": "status-remapper",
                    "enabled": true,
                    "sources": ["level"]
                },
                {
                    "type": "date-remapper",
                    "id": "xn2WAzysQ1asdjJsb9dfb",
                    "enabled": true,
                    "sources": ["timestamp"]
                }
            ]
    }
]
```

## Compatibility with Datadog Pipelines

> CloudPrem processing is designed to align closely with [Datadog Log Log Management][3], allowing direct use of existing pipeline configurations. It achieves this by ignoring unknown or unsupported processors. However, some differences exist:

Reviewer: ```suggestion
CloudPrem processing is designed to align closely with [Datadog Log Management][3], allowing direct use of existing pipeline configurations. It achieves this by ignoring unknown or unsupported processors. However, some differences exist:
```

> - Some filter queries can't be parsed, such as combining wildcards `@data.message:+*`.

Reviewer: ```suggestion
- Some filter queries can't be parsed, such as filters with combined wildcards (for example, `@data.message:+*`).
```
